Sources/Rendering/Core/RenderWindow/index.js

```diff
@@ -34,6 +37,18 @@ function vtkRenderWindow(publicAPI, model) {
 
   publicAPI.hasRenderer = (ren) => model.renderers.indexOf(ren) !== -1;
 
+  // get an API specific view of this data
+  publicAPI.newAPISpecificView = (options) => {
+    if (options && options.api) {
+      return model.apiSpecificViews[options.api]();
+    }
+
+    if (typeof navigator !== 'undefined' && !!navigator.gpu) {
+      return model.apiSpecificViews['WebGPU']();
+    }
+    return model.apiSpecificViews['OpenGL']();
+  }
+
```

Comment:
And thoughts on this file

Reply:
I would write it that way

```
publicAPI.newAPISpecificView = (options={}) => {
  const defaultAPI = navigator.gpu ? 'WebGPU' : 'OpenGL';
  return model.apiSpecificViews[options.api || defaultAPI]
}
```

But also I would not put the `apiSpecificViews` inside the models. This should be private.

Reply:
And you could have `const defaultAPI = navigator.gpu ? 'WebGPU' : 'OpenGL';` be at the top so it get only evaluated at load time.

`const DEFAULT_API = navigator.gpu ? 'WebGPU' : 'OpenGL';`

Reply:
Unless we want to configure it. In which case it could be a field in the model of the RW.

Reply:
You would do a static to query the available APIs (to populate a menu) ?

Reply:
Yes we can list the registered keys as a static method